Auth_phpbb.php

```diff
@@ -504,15 +590,50 @@ public function getCanonicalName( $username )
 
         // Query Database.
         $fresStatement = $fresMySQLConnection->prepare($fstrMySQLQuery);
-        $fresStatement->bind_param('s', $username);
+        $fresStatement->bind_param('s', $this->_phpBBUserName); // bind_param escapes the string
         $fresStatement->execute();
 
         // Bind result
-        $fresStatement->bind_result($resultUsernameClean);
+        $fresStatement->bind_result($resultWikiUsername);
 
         while($fresStatement->fetch())
         {
-            return ucfirst($resultUsernameClean);
+            $this->_wikiUserName = ucfirst($resultWikiUsername); // Preserve capped phpBB username when wikified version is valid
+            return $this->_wikiUserName;
+        }
+
+        // If here, username is invalid or is incompatible with wiki username.
```

Comment:
I can just already imagine the support requests _"I set a wikiusername but my username doesn't get changed?"_

I think it would be best to preserve consistency for the users, even at the cost of a tiny performance hit. I also don't like artificially restricting features that could work for many cases to the one case we imagine it will solve.